### ISSUE TITLE
Embedded Subset Pattern implementation

### ISSUE THREAD

**Issue by etshy** [doctrine/mongodb-odm#2608]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

I learned about this pattern when searching about when/what to embed etc. and I think it could be a great addition to doctrine-odm.
I'm not sure how it could be implemented yet but having an Attribute `#[SubSet(number:5, sort: DESC)]` in addition to an attribute `#[ReferenceMany]` to store the 5 most recents referenced directly on an embed could reduce read operation on the db.

I learned about this pattern on the mongodb website [here](https://www.mongodb.com/basics/embedded-mongodb) and [here](https://www.xuchao.org/docs/mongodb/tutorial/model-embedded-one-to-many-relationships-between-documents.html#subset-pattern)

**Comment by malarzm** [comment]
I believe this resembles #2252 and also suffers the same challenge: tracking changes. ODM would need to know that a (let's say) review has been changed and update its product accordingly.

Also, while less sexy than an attribute, the pattern is already doable and quite elegant:

```
class Product
{
    #[ReferenceMany(targetClass: Review::class)]
    private Collection $reviews;

    #[EmbedMany(targetClass: PartialReview::class)]
    private Collection $recentReviews;

    public function addReview(Review $review)
    {
        $this->reviews[] = $review;
        $this->recentReviews[] = new PartialReview($review);
        // maybe pop one last recent reviews if there's too many
    }
}
```

You could even hide capping recent reviews with a [custom collection](https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/2.5/reference/custom-collections.html) :)

**Comment by etshy** [comment]
Yeah tracking changes and knowing when to fetch the whole references list is what made me think about the most and couldn't think any good solutions for that.
